woocommerce/woocommerce-android
# Top Performers new UI with date range selector

Part of: #11237


### Description

Adds new design to top performers' product. This PR includes:
- Adjust top performers' view to use the new `WidgetCard` composable
- Add new CTA to switch between the different stats granularities independently from dashboard stats view.
- Handle changing custom range. In this first implementation, the top performers view shares the same custom range as the stats view. We agreed on Slack that the custom ranges must be independent of each other. I'm working on it on a different PR, as this PR was already big enough. 
- Adds overflow menu to hide top performers view
- Adds new main CTA to navigate to analytics from top performers' view

This PR doesn't include the new UI for empty cases. We are still waiting for Joe to share updates on this: pe5sF9-2DL-p2#comment-3573


### Testing instructions

1. Enable dynamic dashboard feature flag
2. Test switching different stats granularities and check the stats are updated correctly
3. Check stats granularity works independently from the dashboard stats view. 
4. Set custom range granularity and check top performers' results are loaded correctly for the given custom range. Reminder that the custom range is still shared with the dashboard stats view. Working on changing that.

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/48c6df33-5609-4a3b-8c07-9f4b4a573800
